stats/stats/src/charts/lines/new_operational_txns.rs

```diff
@@ -0,0 +1,141 @@
+use crate::{
+    data_processing::zip_same_timespan,
+    data_source::kinds::{
+        data_manipulation::{
+            map::{Map, MapFunction, MapParseTo, MapToString, StripExt},
+            resolutions::sum::SumLowerResolution,
+        },
+        local_db::{
+            parameters::update::batching::parameters::{
+                Batch30Weeks, Batch30Years, Batch36Months, BatchMaxDays,
+            },
+            DirectVecLocalDbChartSource,
+        },
+    },
+    define_and_impl_resolution_properties,
+    types::{
+        timespans::{Month, Week, Year},
+        Timespan, TimespanValue,
+    },
+    ChartProperties, MissingDatePolicy, Named,
+};
+
+use chrono::NaiveDate;
+use entity::sea_orm_active_enums::ChartType;
+use itertools::Itertools;
+
+use super::{new_blocks::NewBlocksInt, NewTxnsInt};
+
+pub struct Properties;
+
+impl Named for Properties {
+    fn name() -> String {
+        "newOperationalTxns".into()
+    }
+}
+
+impl ChartProperties for Properties {
+    type Resolution = NaiveDate;
+
+    fn chart_type() -> ChartType {
+        ChartType::Line
+    }
+
+    fn missing_date_policy() -> MissingDatePolicy {
+        MissingDatePolicy::FillZero
+    }
+}
+
+pub struct Calculate;
+
+type Input<Resolution> = (
+    // newBlocks
+    Vec<TimespanValue<Resolution, i64>>,
+    // newTxns
+    Vec<TimespanValue<Resolution, i64>>,
+);
+
+impl<Resolution> MapFunction<Input<Resolution>> for Calculate
+where
+    Resolution: Timespan + Send + Ord,
+{
+    type Output = Vec<TimespanValue<Resolution, String>>;
+
+    fn function(inner_data: Input<Resolution>) -> Result<Self::Output, crate::ChartError> {
+        let (blocks_data, txns_data) = inner_data;
+        let combined = zip_same_timespan(blocks_data, txns_data);
+        let data = combined
+            .into_iter()
+            .map(|(timespan, data)| {
+                // both new blocks & new txns treat missing values as zero
+                let (blocks, txns) = data.or(0, 0);
+                TimespanValue {
+                    timespan,
+                    value: txns.saturating_sub(blocks).to_string(),
```

Comment:
I noted that for `TotalOperationalTxns` you used `checked_sub` but here you dont care about negatives, is it intended?

Reply:
arithmetics is done over i64, so it's for `9_223_372_036_854_775_808` rather than just negative numbers

Reply:
fixed for consistency